packages/dds/tree/src/core/tree/mapTree.ts

```diff
@@ -36,17 +39,32 @@ export interface ExclusiveMapTree extends NodeData, MapTree {
  * @privateRemarks This is implemented iteratively (rather than recursively, which is much simpler)
  * to avoid the possibility of a stack overflow for very deep trees.
  */
-export function deepCopyMapTree(mapTree: MapTree): ExclusiveMapTree {
-	type Next = [fields: ExclusiveMapTree["fields"], sourceFields: MapTree["fields"]];
+export function deepCopyMapTree(mapTree: MinimalMapTreeNodeView): ExclusiveMapTree {
```

Comment:
This can now copy from any MinimalMapTreeNodeView: an interface UnhydratedFlexTree nodes start implementing in https://github.com/microsoft/FluidFramework/pull/24739